config.ts

```diff
@@ -0,0 +1,223 @@
+import fs = require('fs');
+import os = require('os');
+import path = require('path');
+
+import base64 = require('base-64');
+import jsonpath = require('jsonpath');
+import request = require('request');
+import shelljs = require('shelljs');
+import yaml = require('js-yaml');
+import api = require('./api');
+
+import client = require('./auth-wrapper');
+
+export class KubeConfig {
+    /**
+     * The list of all known clusters
+     */
+    'clusters': Object[];
```

Comment:
@brendanburns it would be helpful to put specific types on these fields. I wouldn't say that's a blocking factor for this PR. I can do so as a follow-up.